incubator/nodejs/image/project/Dockerfile

```diff
@@ -8,14 +8,29 @@ RUN rm -rf /project/user-app/node_modules && mkdir -p /project/user-app/node_mod
 
 # Install user-app dependencies
 WORKDIR /project/user-app
-RUN npm install --production 
+RUN npm install --production
+
+# Run a build phase. Projects that need to execute build commands can customize the
+# 'build' script in their package.json. The build script should call 'npm install'.
+# If no build is required, just install production dependencies.
+RUN npm run build --if-present
+
+# Uninstall dev dependencies, leaving only production dependencies. Projects can
+# customize the 'prune' script in their package.json.
+# Ideally this would just be 'npm prune', but this command does not have pre/post
+# hooks. Instead, the user's 'prune' script can itself call 'npm prune' in addition
+# to any additional actions required.
+RUN npm run prune --production --if-present
 
 # Creating a tar to work around poor copy performance when using buildah 1.9.0
 RUN cd / && tar czf project.tgz project
 
 # Copy the dependencies into a slim Node docker image
 FROM node:12-slim
 
+# Install common dependencies (TLS and CA support)
+RUN apt-get update && apt-get install -y libssl1.1 ca-certificates && apt-get clean
```

Comment:
In the past we've pulled out any apt get updates/upgrades with a view that we want builds to be reproducible. 

Could we install specific versions of these packages?

Reply:
`FROM node:12-slim` is not reproducible anyway though, is it? That image will be updated regularly by upstream.

Reply:
Similar to what Ian just commented, I'm not sure it would make sense to install more specific versions than these.  This would match (for example) the packages that would be published in the `node:12` (full fat) image.  We'd want to pick up security fixes to all packages automatically, and as I understand it, the way to do this is to regularly republish the image.

If there's a desire to consume those updates in a more selective / scheduled way, I think that's a separate issue and probably needs some tooling around it (like an intermediate slim image that a stack is based on, with its own release / versioning process).

(as an aside, we're also installing these packages in the nodejs-express stack: https://github.com/appsody/stacks/blob/master/incubator/nodejs-express/image/project/Dockerfile#L29 )